### DISCUSSION TITLE
binarySource kubernetes

### DISCUSSION THREAD

**Discussion by micheelengronne** [renovatebot/renovate#22178] [Suggest an Idea]
**What would you like Renovate to be able to do?**
The way `binarySource=Docker` is handled is very docker specific (obviously).

In a kubernetes environment, containers are handled by kube itself. Kube does not necessarily use docker as runtime (it can use CRI-O for instance).

As Kube already handle containers creation and destruction, it can be a great benefit if renovate delegates this behavior (side containers) to kube instead of handling it itself.

**Describe the solution you'd like**
A parameter in renovate to configure delegation when in a kube environment.

**Comment by micheelengronne** [comment]
I can't get deeper in details currently as it is a projection in my future projects (`discussion` state in my backlog).

**Comment by viceice** [comment]
Found this. Seems like a great idea. some suggestions:

* shared read-write-multiple volume
* secrets as file / env
* sidecar pods inherit volumes / secrets
* configure secrets for sidecar container

I'm also intereseted to implement this, as we are migrating to kubernetes. But to be honest, i've not enough time for this at moment. 🙃

**Comment by rarkins** [comment]
What secrets would the sidecar pods need to automatically inherit? I thought they should only get secrets that are explicitly passed by Renovate's artifact logic

**Comment by viceice** [comment]
don't yet know, maybe not neccessary

**Comment by rarkins** [comment]
I think these days kubernetes users would be better off with binarySource=install

**Comment by secustor** [comment]
FYI https://github.com/renovatebot/renovate/issues/19961
